gp/cartesian_graph.py

```diff
@@ -216,17 +222,21 @@ def _format_output_str_of_all_nodes(self):
             for node in active_nodes[hidden_column_idx]:
                 node.format_output_str(self)
 
-    def _fill_parameter_values(self, func_str, parameter_names_to_values):
+    def _fill_parameter_values(
+        self, func_str: str, parameter_names_to_values: Optional[Dict[str, float]] = None
+    ) -> str:
         g = re.findall("<p[0-9]+>", func_str)
-        if g and parameter_names_to_values is None:
+        if len(g) != 0 and parameter_names_to_values is None:
             raise ValueError("parameter node found but no value dict provided")
-        for parameter_name in g:
-            func_str = func_str.replace(
-                parameter_name, str(parameter_names_to_values[parameter_name])
-            )
+        # explicit elif branch is necessary because mypy does not pass otherwise
+        elif len(g) != 0 and parameter_names_to_values is not None:
+            for parameter_name in g:
+                func_str = func_str.replace(
+                    parameter_name, str(parameter_names_to_values[parameter_name])
+                )
```

Comment:
maybe add an explicit `else: pass` branch to indicate that this is not forgotten but intentional to do nothing in that case?

Reply:
Yes, done.